zookeeper-server/src/test/java/org/apache/zookeeper/server/DataTreeTest.java

```diff
@@ -219,7 +219,7 @@ public void testPzxidUpdatedWhenDeletingNonExistNode() throws Exception {
     @Test
     public void testDigestUpdatedWhenReplayCreateTxnForExistNode() {
         try {
-            DigestCalculator.setDigestEnabled(true);
+            System.setProperty(DigestCalculator.ZOOKEEPER_DIGEST_ENABLED, "true");
```

Comment:
This won't have effect of you don't instantiate a new DataTree and the internal DigestCalculator

Reply:
Nice catch, thanks! I modified the test to create a new DataTree after setting the system property.
(I also made the test cleaner by creating the DataTree now only in the test cases and removed the field from the `setUp` method - previously half the tests used the field, half of them not)